ghost.py

```diff
@@ -986,6 +995,130 @@ def _get_document(self, key_name):
+
+    def init(self):
+        """Create a bucket.
+        """
+        self.client.create_bucket(
+            Bucket=self.db_path,
+            CreateBucketConfiguration=self.bucket_configuration)
+
+    def put(self, key):
+        """Insert the key
+        :return: Key name
+        """
+        self.client.put_object(
+            Body=json.dumps(key),
+            Bucket=self.db_path,
+            Key=key['name'])
+        return key['name']
+
+    def list(self):
+        """Lists the keys
+        :return: Returns a list of all keys (not just key names, but rather
+        the keys themselves).
+        """
+        response = self.client.list_objects_v2(Bucket=self.db_path)
+        if u'Contents' in response:
+            # Filter out everything but the key names
+            keys = [key[u'Key'] for key in response[u'Contents']]
+            keys_list = []
+
+            for key_name in keys:
+                key = self.get(key_name)
+                keys_list.append(key)
+
+            return keys_list
+        return []
+
+    def get(self, key_name):
+        """Gets the key.
+        :return: The key itself in a dictionary
+        """
+        def convert_dict_content_to_string(data):
+            """Recursively converts dictionary keys to strings."""
+            if isinstance(data, basestring):
+                return str(data)
+            elif not isinstance(data, dict):
+                return data
+            return dict((str(k), convert_dict_content_to_string(v))
+                        for k, v in data.items())
+
+        try:
+            obj = self.client.get_object(
+                Bucket=self.db_path,
+                Key=key_name)['Body'].read()
+            key = json.loads(obj)
+
+            return convert_dict_content_to_string(key)
+        except botocore.exceptions.ClientError as e:
+            error_code = e.response['Error']['Code']
+            if error_code == 'NoSuchKey':
+                return {}
+
+    def delete(self, key_name):
+        """Delete the key.
+        :return: True if it was deleted, False otherwise
+        """
+        # self.client.delete(key_name)
+        # key = self.get(key_name):
+        # return key is {}
+        self.client.delete_object(
+            Bucket=self.db_path,
+            Key=key_name)
+
+        return self.get(key_name) == {}
+
+    @property
+    def is_initialized(self):
+        """Check if bucket exists.
+        :return: True if initialized, False otherwise
+        """
+        initialized = True
+        try:
+            self.client.head_bucket(Bucket=self.db_path)
+        except botocore.exceptions.ClientError as e:
+            # If a client error is thrown, then check that it was a 404 error.
+            # If it was a 404 error, then the bucket does not exist.
+            error_code = int(e.response['Error']['Code'])
+            if error_code == 404:
+                initialized = False
```

Comment:
Consider reversing this. Initialized is False and then only True if the bucket exists. This way, even if you failed on an unexpected exception, it would fail.